lucene_snapshot: Fix PushQueriesIT.testEqualityAndOther

Fails in the lucene_snapshot branch, with:

```
REPRODUCE WITH: ./gradlew ":x-pack:plugin:esql:qa:server:single-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.single_node.PushQueriesIT.testEqualityAndOther {text}" -Dtests.seed=F22FB909EE7B5F96 -Dtests.locale=sc-IT -Dtests.timezone=Africa/Brazzaville -Druntime.java=24

PushQueriesIT > testEqualityAndOther {text} FAILED
    java.lang.AssertionError: Expected a map containing
                 operator: "LuceneSourceOperator[shards = [test:0], maxPageSize = 4854, remainingDocs = 999]"
                   status: a map containing
          processed_queries: a list containing
                            0: expected ("#test.keyword:vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv -_ignored:test.keyword #foo:[1 TO 1]") but was "#test.keyword:vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv -_ignored:test.keyword"
           processed_slices: <1> unexpected but ok
           processed_shards: <[test:0]> unexpected but ok
               total_slices: <1> unexpected but ok
                  slice_max: <0> unexpected but ok
                    current: <2147483647> unexpected but ok
                  slice_min: <0> unexpected but ok
               rows_emitted: <1> unexpected but ok
              pages_emitted: <1> unexpected but ok
              process_nanos: <6426042> unexpected but ok
    partitioning_strategies: <{test:0=SHARD}> unexpected but ok
                slice_index: <0> unexpected but ok
        at __randomizedtesting.SeedInfo.seed([F22FB909EE7B5F96:846584613070D34F]:0)
        at org.elasticsearch.test.MapMatcher.assertMap(MapMatcher.java:85)
        at org.elasticsearch.test.MapMatcher.assertMap(MapMatcher.java:68)
        at org.elasticsearch.xpack.esql.qa.single_node.PushQueriesIT.checkOperatorProfile(PushQueriesIT.java:338)
        at org.elasticsearch.xpack.esql.qa.single_node.PushQueriesIT.testPushQuery(PushQueriesIT.java:247)
        at org.elasticsearch.xpack.esql.qa.single_node.PushQueriesIT.testEqualityAndOther(PushQueriesIT.java:165)

REPRODUCE WITH: ./gradlew ":x-pack:plugin:esql:qa:server:single-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.single_node.PushQueriesIT.testEqualityAndOther {semantic_text}" -Dtests.seed=F22FB909EE7B5F96 -Dtests.locale=sc-IT -Dtests.timezone=Africa/Brazzaville -Druntime.java=24

PushQueriesIT > testEqualityAndOther {semantic_text} FAILED
    java.net.SocketTimeoutException: 60.000 milliseconds timeout on connection http-outgoing-0 [ACTIVE]
        at __randomizedtesting.SeedInfo.seed([F22FB909EE7B5F96:846584613070D34F]:0)
        at org.elasticsearch.client.RestClient.extractAndWrapCause(RestClient.java:919)
        at org.elasticsearch.client.RestClient.performRequest(RestClient.java:304)
        at org.elasticsearch.client.RestClient.performRequest(RestClient.java:292)
        at org.elasticsearch.xpack.esql.qa.single_node.PushQueriesIT.indexValue(PushQueriesIT.java:326)
        at org.elasticsearch.xpack.esql.qa.single_node.PushQueriesIT.testPushQuery(PushQueriesIT.java:213)
        at org.elasticsearch.xpack.esql.qa.single_node.PushQueriesIT.testEqualityAndOther(PushQueriesIT.java:165)

        Caused by:
        java.net.SocketTimeoutException: 60.000 milliseconds timeout on connection http-outgoing-0 [ACTIVE]
            at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.timeout(HttpAsyncRequestExecutor.java:387)
            at org.apache.http.impl.nio.client.InternalIODispatch.onTimeout(InternalIODispatch.java:98)
            at org.apache.http.impl.nio.client.InternalIODispatch.onTimeout(InternalIODispatch.java:40)
            at org.apache.http.impl.nio.reactor.AbstractIODispatch.timeout(AbstractIODispatch.java:175)
            at org.apache.http.impl.nio.reactor.BaseIOReactor.sessionTimedOut(BaseIOReactor.java:261)
            at org.apache.http.impl.nio.reactor.AbstractIOReactor.timeoutCheck(AbstractIOReactor.java:506)
            at org.apache.http.impl.nio.reactor.BaseIOReactor.validate(BaseIOReactor.java:211)
            at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:280)
            at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
            at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:591)
            at java.base/java.lang.Thread.run(Thread.java:1447)

REPRODUCE WITH: ./gradlew ":x-pack:plugin:esql:qa:server:single-node:javaRestTest" --tests "org.elasticsearch.xpack.esql.qa.single_node.PushQueriesIT.testEquality {semantic_text}" -Dtests.seed=F22FB909EE7B5F96 -Dtests.locale=sc-IT -Dtests.timezone=Africa/Brazzaville -Druntime.java=24

PushQueriesIT > testEquality {semantic_text} FAILED
    java.net.SocketTimeoutException: 60.000 milliseconds timeout on connection http-outgoing-1 [ACTIVE]
        at __randomizedtesting.SeedInfo.seed([F22FB909EE7B5F96:B443561B6955466A]:0)
        at org.elasticsearch.client.RestClient.extractAndWrapCause(RestClient.java:919)
        at org.elasticsearch.client.RestClient.performRequest(RestClient.java:304)
        at org.elasticsearch.client.RestClient.performRequest(RestClient.java:292)
        at org.elasticsearch.xpack.esql.qa.single_node.PushQueriesIT.indexValue(PushQueriesIT.java:326)
        at org.elasticsearch.xpack.esql.qa.single_node.PushQueriesIT.testPushQuery(PushQueriesIT.java:213)
        at org.elasticsearch.xpack.esql.qa.single_node.PushQueriesIT.testEquality(PushQueriesIT.java:85)

        Caused by:
        java.net.SocketTimeoutException: 60.000 milliseconds timeout on connection http-outgoing-1 [ACTIVE]
            at org.apache.http.nio.protocol.HttpAsyncRequestExecutor.timeout(HttpAsyncRequestExecutor.java:387)
            at org.apache.http.impl.nio.client.InternalIODispatch.onTimeout(InternalIODispatch.java:98)
            at org.apache.http.impl.nio.client.InternalIODispatch.onTimeout(InternalIODispatch.java:40)
            at org.apache.http.impl.nio.reactor.AbstractIODispatch.timeout(AbstractIODispatch.java:175)
            at org.apache.http.impl.nio.reactor.BaseIOReactor.sessionTimedOut(BaseIOReactor.java:261)
            at org.apache.http.impl.nio.reactor.AbstractIOReactor.timeoutCheck(AbstractIOReactor.java:506)
            at org.apache.http.impl.nio.reactor.BaseIOReactor.validate(BaseIOReactor.java:211)
            at org.apache.http.impl.nio.reactor.AbstractIOReactor.execute(AbstractIOReactor.java:280)
            at org.apache.http.impl.nio.reactor.BaseIOReactor.execute(BaseIOReactor.java:104)
            at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor$Worker.run(AbstractMultiworkerIOReactor.java:591)
            at java.base/java.lang.Thread.run(Thread.java:1447)
```

--- a/x-pack/plugin/esql/qa/server/single-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/PushQueriesIT.java
+++ b/x-pack/plugin/esql/qa/server/single-node/src/javaRestTest/java/org/elasticsearch/xpack/esql/qa/single_node/PushQueriesIT.java
@@ -10,7 +10,6 @@ package org.elasticsearch.xpack.esql.qa.single_node;
 import com.carrotsearch.randomizedtesting.annotations.ParametersFactory;
 import com.carrotsearch.randomizedtesting.annotations.ThreadLeakFilters;
 
-import org.apache.lucene.tests.util.LuceneTestCase;
 import org.elasticsearch.client.Request;
 import org.elasticsearch.client.Response;
 import org.elasticsearch.client.ResponseException;
@@ -52,7 +51,6 @@ import static org.hamcrest.Matchers.startsWith;
  * Tests for pushing queries to lucene.
  */
 @ThreadLeakFilters(filters = TestClustersThreadFilter.class)
-@LuceneTestCase.AwaitsFix(bugUrl = "https://github.com/elastic/elasticsearch/issues/128506")
 public class PushQueriesIT extends ESRestTestCase {
     @ClassRule
     public static ElasticsearchCluster cluster = Clusters.testCluster(spec -> spec.plugin("inference-service-test"));
@@ -147,8 +145,8 @@ public class PushQueriesIT extends ESRestTestCase {
             | WHERE test == "%value" AND foo == 1
             """;
         List<String> luceneQueryOptions = switch (type) {
-            case "text", "auto" -> List.of("#test.keyword:%value -_ignored:test.keyword #foo:[1 TO 1]");
-            case "match_only_text" -> List.of("foo:[1 TO 1]");
+            case "text", "auto" -> List.of("#test.keyword:%value -_ignored:test.keyword");
+            case "match_only_text" -> List.of("*:*");
             case "semantic_text" ->
                 /*
                  * single_value_match is here because there are extra documents hiding in the index

